### PR TITLE
Add doctor command

### LINKED ISSUE #33: sunodo doctor command
## 📚 Context

Sunodo CLI requires docker to be installed and properly configured.
This is sometimes hard to verify, and errors happen during usage if the system does not meet the minimum requirements.

## ✔️ Solution

Implement a `sunodo doctor` command, similar to `brew doctor` command.
It will verify all the minimal requirements for the proper execution of sunodo commands, tipically:

- check if docker and all necessary subcomponents are installed and running properly
- check the minimal versions of such components

if everything is ok it should print:
```
Your system is ready for sunodo.
```

If it's not ok, it should print the errors.

## 📈 Subtasks

- [x] check if docker is install and running
- [x] check if docker plugins are installed (buildx, compose)
- [x] check if docker can run riscv64 containers

## 🎯 Definition of Done

- [x] print ok if everything is installed and working
- [x] print errors if anything is wrong

### PR DESCRIPTION
**Pull request by oap75**
/close #33

### CODE REVIEW

File: apps/cli/src/commands/doctor.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  static flags = {
+    help: Flags.help({ char: 'h' }),
+  };
+
+  public async run() {
+    try {
+      // Check Docker version
+      this.log('Checking Docker installation...');
+      const dockerVersionOutput = execSync('docker -v').toString().trim();
```
Comment (tuler) [inline apps/cli/src/commands/doctor.ts:17]: Use `docker version --format json` and parse the output.
We need to check for a minimum version (which I honestly don't know yet)

File: apps/cli/src/commands/doctor.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      const dockerVersion = dockerVersionOutput.match(/(\d+\.\d+\.\d+)/)?.[0] || 'Unknown';
+      this.log(`Docker version ${dockerVersion} is installed`);
+
+      // Check Docker Compose version
+      this.log('Checking Docker Compose installation...');
+      const dockerComposeVersionOutput = execSync('docker-compose -v').toString().trim();
```
Comment (tuler) [inline apps/cli/src/commands/doctor.ts:23]: Use `docker compose version --format json`
We need `docker compose`, which is the compose plugin, not the old `docker-compose`

File: apps/cli/src/commands/doctor.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      this.log(`Docker Compose version ${dockerComposeVersion} is installed`);
+
+      // Check Docker Buildx version
+      this.log('Checking Docker Buildx installation...');
+      const dockerBuildxVersionOutput = execSync('docker buildx version').toString().trim();
+      const dockerBuildxVersion = dockerBuildxVersionOutput.match(/(\d+\.\d+\.\d+)/)?.[0] || 'Unknown';
+      this.log(`Docker Buildx version ${dockerBuildxVersion} is installed`);
+
+      // Check Docker riscv64 support
+      this.log('Checking Docker riscv64 support...');
```
Comment (tuler) [inline apps/cli/src/commands/doctor.ts:34]: `docker buildx ls` output was important, it shows `riscv64` support, but it's kind of ugly to parse that.

File: apps/cli/src/commands/doctor.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import { Command, Flags } from '@oclif/core';
+import { execSync } from 'child_process';
```
[the remark this answers is not in the document]
Comment (oap75) [reply apps/cli/src/commands/doctor.ts]: Fixed !

File: apps/cli/src/commands/doctor.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      const dockerBuildxVersionOutput = execSync('docker buildx version').toString().trim();
+      const dockerBuildxVersion = dockerBuildxVersionOutput.match(/(\d+\.\d+\.\d+)/)?.[0] || 'Unknown';
+      this.log(`Docker Buildx version ${dockerBuildxVersion} is installed`);
+
+      // Check Docker riscv64 support
+      this.log('Checking Docker riscv64 support...');
+      execSync('docker run --rm riscv64/hello-world');
+      this.log('Docker riscv64 support is available.');
+
+      this.log('Your system is ready for sunodo.');
```
[the remark this answers is not in the document]
Comment (oap75) [reply apps/cli/src/commands/doctor.ts:91]: Fixed !

File: apps/web/app/page.tsx [hunk 6]
```diff
@@ -24,6 +24,10 @@ const termLines = [
         text: "sunodo shell",
         cmd: true,
     },
+    {
```
[the remark this answers is not in the document]
Comment (oap75) [reply apps/web/app/page.tsx]: Fixed !

File: apps/cli/src/commands/doctor.ts (hunk 1 above)
Comment (oap75) [reply apps/cli/src/commands/doctor.ts]: Fixed ! 

We just need to define minimum docker version.

File: apps/cli/src/commands/doctor.ts (hunk 2 above)
Comment (oap75) [reply apps/cli/src/commands/doctor.ts]: Fixed ! 

We just need to define minimum docker compose plugin version

**Review by tuler — COMMENTED** [review]
It’s getting super nice

File: apps/cli/src/commands/doctor.ts [hunk 7]
```diff
@@ -0,0 +1,100 @@
+import { Command, Flags } from '@oclif/core';
+import { execa } from 'execa';
+
+export default class DoctorCommand extends Command {
+  static description = 'Verify the minimal requirements for the sunodo execution commands';
+
+  static examples = ["<%= config.bin %> <%= command.id %>"];
+
+  static flags = {
+    help: Flags.help({ char: 'h' }),
```
[the remark this answers is not in the document]
Comment (oap75) [reply apps/cli/src/commands/doctor.ts:10]: I don't have for now . 

I just wanted to have this `help flag` to show the sunodo doctor description.

[the remark this answers is not in the document]
Comment (tuler) [reply apps/cli/src/commands/doctor.ts:10]: Help is automatic, you can remove it

File: apps/cli/src/commands/doctor.ts [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+  };
+
+  private static MINIMUM_DOCKER_VERSION = '1.13.0'; // Replace with our minimum required Docker version
+  private static MINIMUM_DOCKER_COMPOSE_VERSION = 'v2.0.0'; // Replace with our minimum required Docker Compose version
+
+  private static compareVersions(v1: string, v2: string): number {
```
[the remark this answers is not in the document]
Comment (oap75) [reply apps/cli/src/commands/doctor.ts]: Fixed !

File: apps/cli/src/commands/doctor.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (oap75) [reply apps/cli/src/commands/doctor.ts]: Fixed !

**Review by tuler — COMMENTED** [review]
Format the code with the prettier plugin

**Comment by tuler** [comment]
Add a [changeset](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md).
In the root dir of the repo run:
```
yarn changeset
```
Select `@sunoco/cli` (using space bar)
Major bump: None (just hit enter)
Minor bump: Select `@sunodo/cli@0.5.0` (using space bar)
Summary: add sunodo doctor command to check system requirements

Commit the file generated in `.changeset`

File: apps/cli/src/commands/doctor.ts [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+import { execa } from "execa";
+import semver from "semver";
+
+export default class DoctorCommand extends Command {
+    static description =
+        "Verify the minimal requirements for the sunodo execution commands";
+
+    static examples = ["<%= config.bin %> <%= command.id %>"];
+
+    private static MINIMUM_DOCKER_VERSION = "1.13.0"; // Replace with our minimum required Docker version
```
Comment (tuler) [inline apps/cli/src/commands/doctor.ts:11]: Let's set the bar high and set this to `23.0.0`

File: apps/cli/src/commands/doctor.ts [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+    private static MINIMUM_DOCKER_COMPOSE_VERSION = "1.0.0"; // Replace with our minimum required Docker Compose version
```
Comment (tuler) [inline apps/cli/src/commands/doctor.ts:12]: Set to `2.16.0`

File: apps/cli/src/commands/doctor.ts (hunk 9 above)
Comment (oap75) [reply apps/cli/src/commands/doctor.ts]: Fixed !

**Comment by tuler** [comment]
LGTM, apply fixup

**Comment by oap75** [comment]
> LGTM, apply fixup

Ready to go !

**Review by tuler — APPROVED** [review]
Good job!

**Pull request merged**